src/passport-saml/xml-utils.ts

```diff
@@ -38,3 +40,50 @@ export const xpath = {
   selectElements: (node: Node, xpath: string): Element[] =>
     selectXPath(elementsXPathTypeGuard, node, xpath),
 };
+
+export const decryptXml = async (xml: string, decryptionKey: string | Buffer) =>
+  util.promisify(xmlenc.decrypt).bind(xmlenc)(xml, { key: decryptionKey });
+
+const normalizeNewlines = (xml: string): string => {
+  // we can use this utility before passing XML to `xml-crypto`
+  // we are considered the XML processor and are responsible for newline normalization
+  // https://github.com/node-saml/passport-saml/issues/431#issuecomment-718132752
+  return xml.replace(/\r\n?/g, "\n");
+};
+
+// This function checks that the |signature| is signed with a given |cert|.
```

Comment:
It might be helpful to use the /** */ notation for this comment so that it appears as part of the function help text in IDEs.